src/api/app/services/workflows/yaml_downloader.rb

```diff
@@ -63,11 +59,24 @@ def download_gitlab_yaml_file
       rescue Gitlab::Error::NotFound => e
         raise Token::Errors::NonExistentRepository, e.message
       end
+      create_temp_file(gitlab_file)
+    end
 
+    def create_temp_file(content)
       tempfile = Tempfile.new(["#{Time.zone.now}", '.yaml'])
-      tempfile.write(gitlab_file)
+      tempfile.write(content)
       tempfile.rewind
       tempfile
     end
+
+    def download_from_url(url)
+      Down.download(url, max_size: MAX_FILE_SIZE)
+    rescue Down::Error => e
+      raise Token::Errors::NonExistentWorkflowsFile, "#{@token.workflow_configuration_url} could not be downloaded.\n#{e.message}" if @token.workflow_configuration_url.present?
+
+      # 'target_branch' can contain a commit sha (when tag push) instead of a branch name
+      raise Token::Errors::NonExistentWorkflowsFile, "#{@token.workflow_configuration_path} could not be downloaded from the SCM branch/commit #{@scm_payload[:target_branch]}." \
+                                                     "\nIs the configuration file in the expected place? Check #{DOCUMENTATION_LINK}\n#{e.message}"
```

Comment:
💭  How do you know that the `Down:Error` means it's not in the right place? It could be any reason...

Reply:
This is not new. It has been the same in master. 
The reasoning is, `Down` is just taking the URL of the file and then just downloads it. If Down is not able to download it then the file does not exist on the given URL. 

It's also important to note that the `Down` gem is only being used for `Gitea` or when the `workflow_configuration_url` is provided in workflow token.